src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -248,14 +242,47 @@ void CRouterRoutino::buildDatabaseList()
                 continue;
             }
 
+            // qDebug() << "buildDatabase" << prefix;
+
 #ifdef Q_OS_WIN
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toLocal8Bit(), prefix.toLocal8Bit());
 #else
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toUtf8(), prefix.toUtf8());
 #endif
+            QVariantMap dmap;
+
             if(data)
             {
-                comboDatabase->addItem(prefix.replace("_", " "), quint64(data));
+                /* determine the profile to use */
+                dmap["db"] = QVariant ((qulonglong)data);
+
+                IAppSetup *setup = IAppSetup::getPlatformInstance();
```

Comment:
Why are you calling getPlatformInstance() on each iteration?

Reply:
Shouldnt be - fixed